pkg/server/backend_manager_test.go

```diff
@@ -426,3 +426,89 @@ func TestDestHostBackendManager_WithDuplicateIdents(t *testing.T) {
 		t.Errorf("expected %v, got %v", e, a)
 	}
 }
+
+func TestParseProxyStrategy(t *testing.T) {
+	for desc, tc := range map[string]struct {
+		input   string
+		want    ProxyStrategy
+		wantErr bool
+	}{
+		"empty": {
+			input:   "",
+			wantErr: true,
+		},
+		"unrecognized": {
+			input:   "unrecognized",
+			wantErr: true,
+		},
+		"default": {
+			input: "default",
+			want:  ProxyStrategyDefault,
+		},
+		"destHost": {
+			input: "destHost",
+			want:  ProxyStrategyDestHost,
+		},
+		"defaultRoute": {
+			input: "defaultRoute",
+			want:  ProxyStrategyDefaultRoute,
+		},
+	} {
+		t.Run(desc, func(t *testing.T) {
+			got, err := ParseProxyStrategy(tc.input)
+			if tc.wantErr != (err != nil) {
+				t.Errorf("ParseProxyStrategy(%s) error: got error %q, want %v", tc.input, err, tc.wantErr)
+			}
+			if err == nil && got != tc.want {
```

Comment:
This seems to test we got a particular value if there is no error.
I think we should validate we get back the right value if say we expected one.
Could we switch this to the following?
`if tc.want != 0 && got != tc.want {`